yurt/yurt_core/provision.py

```diff
@@ -0,0 +1,136 @@
+# coding=utf-8
+
+import click
+from invoke import run
+from yurt.yurt_core.utils import Commands, yes_no_to_bool, echo_multiline
+
+
+# Command Object
+
+
+class ProvisionCommands(Commands):
+    subcommand = 'provision'
+
+
+# Helpers
+
+def aws_ec2(is_dry_run):
+
+    access_key = click.prompt('What is your AWS access key?')
+    key_secret = click.prompt(' What is your AWS secret key?')
+    name = click.prompt('What name will you give this remote host?')
+
+    click.clear()
+
+    invocation = """docker-machine create --driver amazonec2 \\
+    --amazonec2-access-key {} \\
+    --amazonec2-secret-key {} {}""".format(access_key, key_secret, name)
+
+    command = run
+    if is_dry_run:
+        command = echo_multiline
+        command('==> Run this command to provision Docker on a new EC2 Server:')
+
+    command(invocation)
+
+    return name
+
+
+def generic(is_dry_run):
+    ssh_key = click.prompt('What is the path to your ssh key for this host?',
+                           default='~/.ssh/id_rsa',
+                           show_default=True)
+    ssh_ip_address = click.prompt(' What is the IP address for this host?')
+    name = click.prompt('What name will you give this remote host?')
+
+    click.clear()
+
+    invocation = """docker-machine create --driver generic \\
+    --generic-ip-address={} \\
+    --generic-ssh-key {} {}""".format(ssh_ip_address, ssh_key, name)
+
+    command = run
+    if is_dry_run:
+        command = echo_multiline
+        command('==> Run this command to set up Docker on the remote host@{}'.format(ssh_ip_address))
+    command(invocation)
+
+    return name
+
+
+# Commands
+
+def new():
```

Comment:
Do you think this command is how we'd normally setup things for production? I feel like usually there's a lot more to setting up an EC2 instance properly. Is that config stored in the Dockerfile or other related files?

Also curious how this would work with ELB?

Reply:
As far as EC2 specific configuration goes, all of that would live in more options that we'd feed to `docker-machine env --driver amazonec2 ...` ([details here](https://docs.docker.com/machine/drivers/aws/)). If there are some configurations that you know we're going to have to define in every case (`security_group` is one that I can personally attest to), lmk and we can have a conversation about what it would make sense to include OOTB vs. what we can leave to the users/

From a cursory look at ELB + multicontainer capabilities, I came across [this](https://aws.amazon.com/about-aws/whats-new/2015/03/aws-elastic-beanstalk-supports-multi-container-docker-environments/). I'm hesitant to include a `Dockerrun.aws.json` file to this effect for the same reason why I'm not including a Docker Swarm-friendly `docker-compose.yml` file or a Kubernetes config file, which is to say a desire to keep Yurt small, manageable and more focused.